drivers/adc/adc_smartbond_gpadc.c

```diff
@@ -243,20 +280,83 @@ static int adc_smartbond_read_async(const struct device *dev,
 	struct adc_smartbond_data *data = dev->data;
 	int error;
 
+	gpadc_smartbond_pm_policy_state_lock_get(data);
 	adc_context_lock(&data->ctx, true, async);
 	error = start_read(dev, sequence);
 	adc_context_release(&data->ctx, error);
+	gpadc_smartbond_pm_policy_state_lock_put(data);
 
 	return error;
 }
 #endif /* CONFIG_ADC_ASYNC */
 
+#ifdef CONFIG_PM_DEVICE
+static int gpadc_smartbond_resume(const struct device *dev)
+{
+	int ret;
+	struct adc_smartbond_data *data = dev->data;
+	const struct adc_smartbond_cfg *config = dev->config;
+
+	GPADC->GP_ADC_CTRL_REG = data->gpadc_ctrl_reg;
+
+	/*
+	 * Configure dt provided device signals when available.
+	 * pinctrl is optional so ENOENT is not setup failure.
+	 */
+	ret = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0 && ret != -ENOENT) {
+		LOG_ERR("ADC pinctrl setup failed (%d)", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int gpadc_smartbond_suspend(const struct device *dev)
+{
+	int ret;
+	struct adc_smartbond_data *data = dev->data;
+	const struct adc_smartbond_cfg *config = dev->config;
+
+	data->gpadc_ctrl_reg = GPADC->GP_ADC_CTRL_REG;
+	/* Disable the GPADC LDO */
+	GPADC->GP_ADC_CTRL_REG = 0;
+
+	ret = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_SLEEP);
+	if (ret < 0) {
+		LOG_WRN("Failed to configure the GPADC pins to inactive state");
+	}
+
+	return ret;
+}
+
+static int gpadc_smartbond_pm_action(const struct device *dev,
+				   enum pm_device_action action)
+{
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		da1469x_pd_acquire(MCU_PD_DOMAIN_PER);
+		gpadc_smartbond_resume(dev);
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+		gpadc_smartbond_suspend(dev);
+		da1469x_pd_release(MCU_PD_DOMAIN_PER);
+		break;
+	default:
+		return -ENOTSUP;
+	}
+	return 0;
+}
+#endif /* CONFIG_PM_DEVICE */
+
 static int adc_smartbond_init(const struct device *dev)
 {
 	int err;
 	struct adc_smartbond_data *data = dev->data;
 	const struct adc_smartbond_cfg *config = dev->config;
 
+	da1469x_pd_acquire(MCU_PD_DOMAIN_PER);
+
 	GPADC->GP_ADC_CTRL_REG = GPADC_GP_ADC_CTRL_REG_GP_ADC_EN_Msk;
```

Comment:
Following my previous comment on the device runtime scheme, ADC registers should be handled before sampling takes place, that should be in `adc_context_start_sampling`.

Reply:
I moved the register setup to adc_context_start_sampling